dozer-ingestion/src/connectors/postgres/helper.rs

```diff
@@ -147,14 +155,10 @@ pub fn value_to_field(
             let value: Result<Vec<u8>, _> = row.try_get(idx);
             value.map_or_else(handle_error, |v| Ok(Field::Binary(v)))
         }
-        &Type::JSONB => {
-            let value: Result<serde_json::Value, _> = row.try_get(idx);
-
-            value.map_or_else(handle_error, |v| {
-                Ok(Field::Bson(
-                    bson::to_vec(&v).map_err(|e| JSONBParseError(e.to_string()))?,
-                ))
-            })
+        &Type::JSONB | &Type::JSON => convert_row_value_to_field!(row, idx, serde_json::Value),
+        &Type::JSONB_ARRAY | &Type::JSON_ARRAY => {
```

Comment:
because `*_ARRAY` type cannot be natively converted to `serde_json::Value` as the macro that we are using is like followed:

```rust
macro_rules! convert_row_value_to_field {
    ($a:ident, $b:ident, $c:ty) => {{
        let value: Result<$c, _> = $a.try_get($b);
        value.map_or_else(handle_error, |val| Ok(Field::from(val)))
    }};
}
```
what we need for `*_ARRAY` is `Vec<serde_json::Value>` so had to parse differently. However, in `postgres_type_to_field` we don't specify `serde_json::Value` while deserialize from bytes using `from_slice`. In this case, it automatically gets its type as `Vec<serde_json::Value>` in case it is `*_ARRAY`.

Reply:
Ah got your point, then `postgres_type_to_field` need to be changed, because in case of change this here itself as followed, 

```rust
 &Type::JSONB | &Type::JSON | &Type::JSONB_ARRAY | &Type::JSON_ARRAY => {
            convert_row_value_to_field!(row, idx, serde_json::Value)
        }
```
then will get this error 

```
ERROR Value conversion error: cannot convert between the Rust type `serde_json::value::Value` and the Postgres type `_jsonb`
```

Reply:
I still don't get it. If some bytes can be successfully deserialized to `Vec<serde_json::Value>` in JSON format, then by definition it's also a `serde_json::Value`, because an array of JSON value is a JSON value. Can you try to find the root cause of the failure?

Reply:
yes the cause is due to

```rust
let value: Result<$c, _> = $a.try_get($b);
```

in the macro, it cannot be getting try_get a row which is serde_json::Value which is already an Array type

Reply:
Got it! Thanks!